pkg/analyzer/pvc_test.go

```diff
@@ -0,0 +1,217 @@
+		expectations []string
+	}{
+		{
+			name: "PV1 and PVC5 report failures",
+			config: common.Analyzer{
+				Client: &kubernetes.Client{
+					Client: fake.NewSimpleClientset(
+						&appsv1.Event{
+							// This is the latest event.
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "Event1",
+								Namespace: "default",
+							},
+							Reason:  "ProvisioningFailed",
+							Message: "PVC provisioning failed",
+						},
+						&appsv1.Event{
+							ObjectMeta: metav1.ObjectMeta{
+								// This event won't get selected.
+								Name:      "Event2",
+								Namespace: "test",
+							},
+						},
+						&appsv1.Event{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "Event3",
+								Namespace: "default",
+							},
+						},
+						&appsv1.PersistentVolumeClaim{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "PVC1",
+								Namespace: "default",
+							},
+							Status: appsv1.PersistentVolumeClaimStatus{
+								Phase: appsv1.ClaimPending,
+							},
+						},
+						&appsv1.PersistentVolumeClaim{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "PVC2",
+								Namespace: "default",
+							},
+							Status: appsv1.PersistentVolumeClaimStatus{
+								// Won't contribute to failures.
+								Phase: appsv1.ClaimBound,
+							},
+						},
+						&appsv1.PersistentVolumeClaim{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "PVC3",
+								Namespace: "default",
+							},
+							Status: appsv1.PersistentVolumeClaimStatus{
+								// Won't contribute to failures.
+								Phase: appsv1.ClaimLost,
+							},
+						},
+						&appsv1.PersistentVolumeClaim{
+							// PVCs in namespace other than "default" won't be discovered.
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "PVC4",
+								Namespace: "test",
+							},
+							Status: appsv1.PersistentVolumeClaimStatus{
+								Phase: appsv1.ClaimLost,
+							},
+						},
+						&appsv1.PersistentVolumeClaim{
+							// PVCs in namespace other than "default" won't be discovered.
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "PVC5",
+								Namespace: "default",
+							},
+							Status: appsv1.PersistentVolumeClaimStatus{
+								Phase: appsv1.ClaimPending,
+							},
+						},
+					),
+				},
+				Context:   context.Background(),
+				Namespace: "default",
+			},
+			expectations: []string{
+				"default/PVC1",
+				"default/PVC5",
+			},
+		},
+		{
+			name: "no event",
+			config: common.Analyzer{
+				Client: &kubernetes.Client{
+					Client: fake.NewSimpleClientset(
+						&appsv1.PersistentVolumeClaim{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "PVC1",
+								Namespace: "default",
+							},
+							Status: appsv1.PersistentVolumeClaimStatus{
+								Phase: appsv1.ClaimPending,
+							},
+						},
+					),
+				},
+				Context:   context.Background(),
+				Namespace: "default",
+			},
+		},
+		{
+			name: "event other than privision failure",
```

Comment:
just a small typo

Reply:
Oops, thanks for reviewing and pointing this out. I'll take care of it in the upcoming PRs.